Scripts/BuildDatabank/AddData.py

```diff
@@ -16,572 +35,550 @@
+    # of mass the membrane each the centre of mass of a lipid molecule is.
+    # If a lipid molecule is split so that headgroup and tails are their own residues, the centre of mass of the
+    # headgroup is used in the calculation.
+    ################################################################################################################
 
-## Check that the number of atoms between data and README.yaml match
+    logger.info(
+        "Calculating the numbers of lipid molecules in each leaflet based on the center of mass of the membrane and lipids."
+    )
+    logger.info(
+        "If a lipid molecule is split to multiple residues, the centre of mass of the headgroup is used."
+    )
 
-number_of_atomsTRJ = len(u.atoms)
+    top = ""
+    traj = ""
 
-number_of_atoms = 0
-for key_mol in all_molecules:
-    mapping_dict = {}
-    try:
-        mapping_file = os.path.join(
-            os.getcwd(), "mapping_files", sim["COMPOSITION"][key_mol]["MAPPING"]
-        )
-    except:
-        continue
+    if sim["SOFTWARE"] == "gromacs":
+        top = os.path.join(dir_tmp, sim["TPR"][0][0])
+        traj = os.path.join(dir_tmp, sim["TRJ"][0][0])
+    elif sim["SOFTWARE"] == "openMM" or sim["SOFTWARE"] == "NAMD":
+        traj = os.path.join(dir_tmp, sim["TRJ"][0][0])
+        top = os.path.join(dir_tmp, sim["PDB"][0][0])
     else:
-        with open(mapping_file, "r") as yaml_file:
-            mapping_dict = yaml.load(yaml_file, Loader=yaml.FullLoader)
-        yaml_file.close()
+        logger.error(
+        "SOFTWARE '%s' is not a proper option.\n"
+        "Use either 'gromacs', 'openMM', or 'NAMD'.")
+        quit()
 
-    if sim.get("UNITEDATOM_DICT") and not "SOL" in key_mol:
-        mapping_file_length = 0
+    leaflet1 = 0  # total number of lipids in upper leaflet
+    leaflet2 = 0  # total number of lipids in lower leaflet
 
-        for key in mapping_dict.keys():
-            if "H" in key:
-                continue
-            else:
-                mapping_file_length += 1
 
-    else:
-        mapping_file_length = len(mapping_dict.keys())
+    gro = os.path.join(dir_tmp, "frame0.gro")
+    NewTraj = os.path.join(dir_tmp, "NewTraj.xtc")
 
     try:
-        number_of_atoms += (
-            np.sum(sim["COMPOSITION"][key_mol]["COUNT"]) * mapping_file_length
+        logger.info(f"MDAnalysis tries to use {top} and {traj}")
+        u = Universe(top, traj)
+        u.atoms.write(gro, frames=u.trajectory[[0]])  # write first frame into gro file
+    except Exception as e:
+        logger.warning(e)
+        logger.info(
+            "Now generating frame0.gro with Gromacs because MDAnalysis cannot read tpr version ..."
         )
+        if ( "WARNINGS" in sim and 
+            sim["WARNINGS"] is not None and
+            sim["WARNINGS"]["GROMACS_VERSION"] == "gromacs3" ):
+            logger.debug(
+                f"executing 'echo System | gmx trjconv -s {top} -f {traj} -dump 22000 -o {gro}'"
```

Comment:
no idea. Some heritage. Changed gmx trjconv => trjconv, 22000 => 0

Reply:
I think it is an ad hock for a single trajectory. I am inclined to either remove the trajectory from the databank, or come up with a more general solution.

Reply:
Here is the list of the trajectories with `GROMACS_VERSION: gromacs3`
```
0a2/272/0a22727a0659852772b4a1193ced99c5981fb739/973cf49703e3217f44b5c18759fd8926e8f5d1f1
219/709/219709a1090be0c8ecc1fc3c6362cdf996c1c997/fdf916cac7bed9521a8d6acd71f12207aa12f6fd
52e/142/52e1425c86de88b82dc50fc6539269b4e73b3a61/cac0f8f72a05076af7c53c7a17bf05f20666c5d7
532/695/5326957d67eb0355d0eccf6aac9d9a44fd984777/48e776c91986eeb05d8d407567423c42bbc6126e
671/199/671199c998a93dcf594469ea33e3b25e73d08b9e/25bf45d5d477c8ae86de3b9c044ead9489d678f6
7ab/2b4/7ab2b45dcff6f8ff34f5f26d8d15fb94f7cf3393/1d42d856ac5622426ba19d7a73030b1a749f817a
7cb/7a3/7cb7a3281b5e79d8e0328d82cdebbcc11f1bab2a/f89b798e078bcc3f9793b20176dbe25ae42778ad
88c/02d/88c02d2e58d6c59918f230e060130570e398071e/32ba461564dfbf8212488be80ddf109e3de23c1f
a69/b1f/a69b1f89e3c9dd265a7ac3009eceb75695de582b/51b9a8cb6e4f8c4f17816bfcc025978b93276e5d
f78/e18/f78e188379aa6f6b1be9f2a8afc8bc2aac191ed0/443765a3de567fd6fe4e8666d29c63e8580e6306
f97/dff/f97dfff6e14f18c3d24bda5b8fdf9d4b4a7ec763/d9fa5dfeae8940ba2ee2faa7784bec5b2d44613a
```

Do you want me to open an issue about the discussion of supporting these records?

Reply:
I want to avoid using 22000 magic number and also the gromacs 3 dependency. So yes, maybe an issue can be worth.

I believe that 22000 is just a magic number that once was established for all this trajectories. We actually can't use MDAnalysis or other tools to extract the value of the first frame. What can be done, is application of `gmx check` on the trajectory and then parsing the output to understand what is the time of the first frame. Then using this time as a parameter for `dump`. It is not really complicated, but maybe it is too much work for supporting 10 trajectories.

Reply:
I will mention this in the issue that already exists, and it is related: https://github.com/NMRLipids/Databank/issues/195
Please look inside. Probably we should just avoid having non-zero timestep for further use.